osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -445,6 +453,32 @@ public virtual void PopulateScore(ScoreInfo score)
             score.TotalScore = (long)Math.Round(ComputeFinalScore(ScoringMode.Standardised, score));
         }
 
+        /// <summary>
+        /// Populates the given score with remaining statistics as "missed" and marks it with <see cref="ScoreRank.F"/> rank.
+        /// </summary>
+        public void FailScore(ScoreInfo score)
+        {
+            if (Rank.Value == ScoreRank.F)
+                return;
+
+            Rank.Value = ScoreRank.F;
+            Rank.Disabled = true;
```

Comment:
Was this done somewhere else before? Not sure about this. It probably won't handle rewinding too well.

Reply:
I've disabled the bindable so that the rank doesn't silently change back and potentially regress showing failed scores with F rank, but rewinding and reverting failure is an interesting point. I guess I'll remove it for now.